doc/designs/prci_checker.md

```diff
@@ -0,0 +1,149 @@
+# FreeIPA Pull Request CI (PR-CI) checker tool
+
+[PR-CI checker](https://github.com/freeipa/freeipa/blob/master/ipatests/prci_definitions/prci_checker.py) is a tool which checks that FreeIPA PR-CI definition files follow a pre-defined convention such as having the correct naming format, requirements and arguments.
+
+Usual usecase is to check the format of newly added jobs in the PR-CI definitions
+
+## Basic usage
+
+**To check all the definitions**:
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] nightly_previous [Jobs] 149
+[File] nightly_latest_pki [Jobs] 75
+[File] gating [Jobs] 24
+[File] nightly_rawhide [Jobs] 149
+[File] nightly_latest_389ds [Jobs] 45
+[File] nightly_latest [Jobs] 149
+[File] nightly_latest_selinux [Jobs] 149
+[File] nightly_latest_testing [Jobs] 149
+[File] nightly_latest_sssd [Jobs] 16
+[File] nightly_latest_testing_selinux [Jobs] 149
+CHECKS FINISHED SUCCESSFULLY
+```
+
+**alternatively:**
+
+``` bash
+$ make yamllint
+[...]
+Check PRCI definitions
+-----------
+/usr/bin/python ./ipatests/prci_definitions/prci_checker.py -d ./ipatests/prci_definitions -s ./ipatests/prci_definitions/prci_jobs_spec.yaml;
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] nightly_previous [Jobs] 149
+[...]
+[File] nightly_latest_testing_selinux [Jobs] 149
+CHECKS FINISHED SUCCESSFULLY
+-----------
+```
+
+**To check a specific file:**
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/gating.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] gating [Jobs] 24
+CHECKS FINISHED SUCCESSFULLY
+```
+
+## Tool specification file
+
+Most part of the checks carried out by the script are described in the specification file [prci_jobs_spec.yaml](https://github.com/freeipa/freeipa/blob/master/ipatests/prci_definitions/prci_jobs_spec.yaml).
+
+The yaml file contains suppported classes, definitions that should have the same number of jobs, and format specification for each file.
+
+**File content can vary according to the different needs in every branch**. [Example for `ipa-4-10`](https://github.com/freeipa/freeipa/blob/ipa-4-10/ipatests/prci_definitions/prci_jobs_spec.yaml):
+
+``` yaml
+# Specification file for PRCI definitions used by prci_checker script
+
+# List of supported test-run classes for non-build jobs
+classes:
+  - "RunPytest"
+  - "RunPytest2"
+  - "RunPytest3"
+  - "RunWebuiTests"
+  - "RunADTests"
+
+# (Optional) Definition files that should contain the same number of jobs
```

Comment:
Here and/or in the original file it might be nice to document why there are definitions that should contain the same number of jobs. I assume it is so we don't accidentally miss something.

Reply:
I would be happy to add the reason behind this check, although I don't usually work with PR-CI so I'm not sure. 
Probably @flo-renaud can help us here.

Reply:
The prci_definitions directory contains multiple test definition files:
- for gating: only a subset of tests, the ones that are executed for each pull request
- for temp commit: this is a template allowing to run a single test, usually to demonstrate that changes done to this test are working.
- for nightly tests. There are some general purpose nightlies (for instance on the master branch, we have a nightly run for fedora latest, another one for fedora previous, another one with selinux etc...) or more specialized ones, to test integration with the latest code of specific components (389ds, sssd or pki). The general purpose nightlies execute all the tests, whereas the specific nightlies execute only a subset of tests, the ones more relevant for the component.

This check ensures that a new test gets added to all the general purpose nightlies.

Reply:
Thank you for you insight @flo-renaud 
I have integrated this info into the doc